File: python/core/foldset/web.py
from __future__ import annotations

from .paywall import generate_paywall_html
from .types import PaymentMethod, ProcessRequestResult, RequestAdapter, WebRestriction


def format_web_payment_error(
    result: ProcessRequestResult,
    restriction: WebRestriction,
    payment_methods: list[PaymentMethod],
    adapter: RequestAdapter,
    terms_of_service_url: str | None = None,
) -> None:
    result.response.body = generate_paywall_html(
        restriction, payment_methods, adapter.get_url(), terms_of_service_url
    )
    result.response.headers["Content-Type"] = "text/html"

File: typescript/core/src/web.ts
import { generatePaywallHtml } from "./paywall";
import type { PaymentMethod, ProcessRequestResult, RequestAdapter, WebRestriction } from "./types";

export function formatWebPaymentError(
  result: Extract<ProcessRequestResult, { type: "payment-error" }>,
  restriction: WebRestriction,
  paymentMethods: PaymentMethod[],
  adapter: RequestAdapter,
  termsOfServiceUrl?: string,
): void {
  result.response.body = generatePaywallHtml(restriction, paymentMethods, adapter.getUrl(), termsOfServiceUrl);
  result.response.headers["Content-Type"] = "text/html";
}
